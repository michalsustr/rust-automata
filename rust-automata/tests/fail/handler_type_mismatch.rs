use rust_automata::*;

// States
#[derive(Default)]
pub struct S1;
#[derive(Default)]
pub struct S2;

// Input
#[derive(Default)]
pub struct I1;

#[automaton(
    inputs(I1),
    states(S1, S2),
    outputs(),
    transitions(
        (S1, I1) -> (S2) = handle_transition
    )
)]
pub struct HandlerTypeMismatch;

impl HandlerTypeMismatch {
    // Should return S2
    fn handle_transition(&mut self, _: S1, _: I1) -> bool { 
        true
    }
}

fn main() {}