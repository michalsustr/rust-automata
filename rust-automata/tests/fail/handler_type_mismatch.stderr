error: cannot find attribute `automaton` in this scope
  --> tests/fail/handler_type_mismatch.rs:13:3
   |
13 | #[automaton(
   |   ^^^^^^^^^

warning: unused import: `rust_automata::*`
 --> tests/fail/handler_type_mismatch.rs:1:5
  |
1 | use rust_automata::*;
  |     ^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
